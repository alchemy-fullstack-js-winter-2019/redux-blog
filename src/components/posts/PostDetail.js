import React from 'react';
import PropTypes from 'prop-types';
import Comments from '../comments/Comments';

export default function PostDetail({ post, comments }) {
  return (
    <>
      <h2>{post.title}</h2>
      <p>{post.body}</p>
      <h4>Comments</h4>
      <Comments comments={comments}/>
    </>
  );
}

PostDetail.propTypes = {
  post: PropTypes.object,
  comments: PropTypes.array
};
