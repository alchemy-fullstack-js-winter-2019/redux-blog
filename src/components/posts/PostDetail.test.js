import React from 'react';
import renderer from 'react-test-renderer';
import PostDetail from './PostDetail';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Post', () => {
  it('matches a snapshot', () => {
    const post = {
      title: 'title',
      body: 'body'
    };

    const comments = [
      {
        title: 'title1',
        body: 'body1'
      },
      {
        title: 'title2',
        body: 'body2'
      }
    ];

    const user = {
      name: 'name',
      email: 'email'
    };

    const tree = renderer.create(
      <Router>
        <PostDetail comments={comments} post={post} user={user} />
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
