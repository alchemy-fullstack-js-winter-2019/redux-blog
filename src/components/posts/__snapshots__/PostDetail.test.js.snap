// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post matches a snapshot 1`] = `
Array [
  <h2>
    title
  </h2>,
  <p>
    body
  </p>,
  <h4>
    Comments
  </h4>,
  <ul>
    <li>
      <p>
        body1
      </p>
      <p>
        Email: 
      </p>
    </li>
    <li>
      <p>
        body2
      </p>
      <p>
        Email: 
      </p>
    </li>
  </ul>,
]
`;
